### ISSUE TITLE
collapse hides new-lines between code blocks

### ISSUE THREAD

**Issue by artin-phares** [Microsoft/vscode#36145]
- VSCode Version: 1.16.1
- OS Version: windows 7

Hi, small annoying issue with code editor (javascript).  

1. Create two function calls like this:
```
callA(
    arg1,
    arg2);

callB();
```
2. Collapse first call  

actual (new line is gone):
```
callA(...
callB();
```

expected (new line is in place):
```
callA(...

callB();
```

Adding any number of new-lines between calls does nothing, `collapse` will eat them all.  

real world case:  

Noticed while writing tests, where I have long test description on the first line, and function on second line (because of line length limit).  While collapsing I need to see test description, and have a space between collapsed tests. Space between improves visibility (1), and allows to move/delete individual tests (2). Currently when selecting collapsed block it selects only visible header, but not hidden part (which is another issue probably).

demo:  

![gif](https://user-images.githubusercontent.com/671082/31504385-eb14899a-af7a-11e7-8aef-f90291e3c8b6.gif)

**Comment by aeschli** [comment]
That's how indentation based folding works. In JavaScript, new lines belong to the preceding block.
The motivation is to optimize for code like:
```
function f() {
  //some code that will be folded

}
```

To solve both requirements, language awareness is needed. The issue for that is #3422

**Comment by artin-phares** [comment]
So, current behavior for javascript will stay because it's expected behavior, or it will be fixed by #3422?  

btw, I believe this is the same for Python https://github.com/Microsoft/vscode/issues/3353

**Comment by aeschli** [comment]
Python is different: there the empty lines should not be folded. That got fixed by #3353.
Yes, only once we have #3422 we can be smarter in Javadoc and also support your example.
